### PR TITLE
Add all_parameters api for Program

### PR DESCRIPTION
**Pull request by chenwhql**
add `all_parameters` interface for `fluid.Program`, so that users no longer need to use `Program.global_block().all_parameters()` to get parameters of Program.

image

cn doc: https://github.com/PaddlePaddle/FluidDoc/pull/1724

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_program.py [hunk 1]
```diff
@@ -132,6 +132,16 @@ def net():
...
             self.assertEqual(no_read_ops[i].type, keep_read_ops[i + 2].type)
 
+    def test_program_all_parameters(self):
+        program = fluid.default_main_program()
+        data = fluid.data(name='x', shape=[None, 13], dtype='float32')
+        hidden = fluid.layers.fc(input=data, size=10)
+        loss = fluid.layers.mean(hidden)
+        fluid.optimizer.SGD(learning_rate=0.01).minimize(loss)
+
+        for param in program.all_parameters():
```
Comment (sneaxiy) [inline python/paddle/fluid/tests/unittests/test_program.py]: Add some assertion? For example, `self.assertEqual(len(program.all_parameters()), xxx)`.

Comment (chenwhql) [reply python/paddle/fluid/tests/unittests/test_program.py]: Done

**Review by sneaxiy — DISMISSED** [review]

**Review by Boyan-Liu — DISMISSED** [review]
LGTM

File: python/paddle/fluid/framework.py [hunk 2]
```diff
@@ -4524,6 +4524,33 @@ def list_vars(self):
...
+                import paddle.fluid as fluid
+
+                program = fluid.default_main_program()
+                data = fluid.data(name='x', shape=[None, 13], dtype='float32')
+                hidden = fluid.layers.fc(input=data, size=10)
+                loss = fluid.layers.mean(hidden)
+                fluid.optimizer.SGD(learning_rate=0.01).minimize(loss)
+
+                for param in program.all_parameters():
+                    print(param)
```
Comment (lanxianghit) [inline python/paddle/fluid/framework.py:4547]: 请加上打印结果，以及对结果的一个简要说明

Comment (chenwhql) [reply python/paddle/fluid/framework.py:4547]: Done

**Review by lanxianghit — APPROVED** [review]
LGTM

**Review by sneaxiy — APPROVED** [review]

**Review by Boyan-Liu — APPROVED** [review]

**Pull request merged**